napari_clusters_plotter/_tests/test_plotter.py

```diff
@@ -79,6 +79,278 @@ def test_plotting(make_napari_viewer):
     print("check")
 
 
+def test_plotter_on_labels3d(make_napari_viewer):
+    from napari_clusters_plotter._plotter import PlotterWidget
+    from napari_clusters_plotter._utilities import get_layer_tabular_data
+
+    viewer = make_napari_viewer()
+    # Generate 3D data
+    z_slices = 3
+    n_labels = get_labels_array().max()
+    data_3d = np.stack([get_labels_array() for _ in range(z_slices)])
+    # Create some random features
+    label_column = np.arange(1, n_labels + 1)
+    feature1 = np.random.normal(size=n_labels)
+    feature2 = np.random.normal(size=n_labels, loc=1)
+    annotations = 1 * (np.random.uniform(size=n_labels) > 0.5)
+
+    viewer.add_labels(
+        data_3d,
+        properties={
+            "label": label_column,
+            "feature1": feature1,
+            "feature2": feature2,
+            "annotations": annotations,
+        },
+    )
+
+    widget = PlotterWidget(viewer)
+    viewer.window.add_dock_widget(widget)
+
+    # Put the labels layer into the layer selection widget
+    widget.analysed_layer = viewer.layers[0]
+    widget.plot_x_axis.setCurrentText("feature1")
+    widget.plot_y_axis.setCurrentText("feature2")
+    widget.analysed_layer.features["MANUAL_CLUSTER_ID"] = annotations
+
+    # check that the features are found
+    features = get_layer_tabular_data(widget.analysed_layer)
+    assert "feature1" in features.columns
+    assert "feature2" in features.columns
+
+    widget.run(
+        widget.analysed_layer.features,
+        plot_x_axis_name="feature1",
+        plot_y_axis_name="feature2",
+        plot_cluster_name="MANUAL_CLUSTER_ID",
+        force_redraw=True,
+    )
+
+
+def test_plotter_on_labels2d_timelapse(make_napari_viewer):
+    from napari_clusters_plotter._plotter import PlotterWidget
+    from napari_clusters_plotter._utilities import get_layer_tabular_data
+
+    viewer = make_napari_viewer()
+    # Generate 2D timelapse data
+    n_timepoints = 2
+    n_labels = get_labels_array().max()
+    data_2d_timelapse = np.array(
+        [np.roll(get_labels_array(), t, axis=0) for t in range(n_timepoints)]
+    )
+    data_2d_timelapse = np.expand_dims(
+        data_2d_timelapse, axis=1
+    )  # add unidimensional z axis
+    # Create some random features
+    label_column = np.tile(np.arange(1, n_labels + 1), n_timepoints)
+    feature1 = np.random.normal(size=n_labels * n_timepoints)
+    feature2 = np.random.normal(size=n_labels * n_timepoints, loc=1)
+    annotations = 1 * (np.random.uniform(size=n_labels * n_timepoints) > 0.5)
+    frame = np.arange(0, n_timepoints).repeat(
+        n_labels
+    )  # add frame column for timelapsed data
+
+    viewer.add_labels(
+        data_2d_timelapse,
+        properties={
+            "label": label_column,
+            "feature1": feature1,
+            "feature2": feature2,
+            "annotations": annotations,
+            "frame": frame,
+        },
+    )
+
```

Comment:
```suggestion
    assert len(data_3d_timelapse) == 4
```

Reply:
Here it should be `data_2d_timelapse`. I will fix that.